### PR TITLE
Remove unecessary step from release docs

### PR DESCRIPTION
**Pull request by astrofrog**
This paragraph is out of date - new version tags are activated automatically.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/development/maintainers/releasing.rst [hunk 1]
```diff
@@ -429,12 +429,11 @@ clean-up tasks to finalize the process.
 Post-Release procedures
 -----------------------
 
-#. Update Readthedocs so that it builds docs for the version you just released.
-   You'll find this in the "Admin" tab, in the "Edit Versions" section --
-   click on "Activate" for the tag of the release you have just done.
-   Also verify that the ``stable`` Readthedocs version builds correctly for
-   the new version (it should trigger automatically once you've done the
-   previous step).
+#. Make sure that Readthedocs is building the documentation for the version you just released.
```
Comment (pllim) [inline docs/development/maintainers/releasing.rst:432]:
```suggestion
#. Make sure that ReadTheDocs is building the documentation for the version you just released.
```

File: docs/development/maintainers/releasing.rst [hunk 2]
```diff
@@ -429,12 +429,11 @@ clean-up tasks to finalize the process.
...
+   You'll find this in the "Versions" tab -- click on "Edit" for the tag you
+   just released, and make sure that "Active" is checked. Also verify that the
+   ``stable`` Readthedocs version builds correctly for the new version (it
```
Comment (pllim) [inline docs/development/maintainers/releasing.rst:435]:
```suggestion
   ``stable`` ReadTheDocs version builds correctly for the new version (it
```

**Review by pllim — APPROVED** [review]
Might as well capitalize properly?

Thanks!

**Review by saimn — APPROVED** [review]
LGTM

**Pull request merged**